java/cdk/src/main/java/sleeper/cdk/stack/TransactionLogSnapshotStack.java

```diff
@@ -130,7 +144,73 @@ public TransactionLogSnapshotCreationStack(
         coreStacks.grantInvokeScheduled(snapshotCreationTrigger, queue);
         coreStacks.grantReadTablesStatus(snapshotCreationLambda);
         transactionLogStateStoreStack.grantReadWriteSnapshots(snapshotCreationLambda);
+    }
 
-        Utils.addStackTagIfSet(this, instanceProperties);
+    private void createSnapshotDeletionLambda(InstanceProperties instanceProperties, LambdaCode statestoreJar, CoreStacks coreStacks,
+            TransactionLogStateStoreStack transactionLogStateStoreStack, Topic topic, List<IMetric> errorMetrics) {
+        String triggerFunctionName = Utils.truncateTo64Characters(String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "transaction-log-snapshot-deletion-trigger"));
```

Comment:
This name is too long, it comes to 70 characters with a 20 character instance ID. We should probably avoid using Utils.truncateTo64Characters. We can avoid the limit given we've limited the length of the instance ID.

Reply:
I've raised a couple of issues related to this:

- https://github.com/gchq/sleeper/issues/2451
- https://github.com/gchq/sleeper/issues/2452

Reply:
Never mind, this is within the limit for SQS. I thought all resources were limited at 64 characters.

Reply:
Lambda function names are limited to 64 characters:

https://docs.aws.amazon.com/lambda/latest/api/API_CreateFunction.html#API_CreateFunction_RequestBody